xxh_arm64sve.S

```diff
@@ -0,0 +1,475 @@
+/*
+ * xxHash - Extremely Fast Hash algorithm
+ *
+ * Copyright (C) 2022 Linaro Ltd.
+ *
+ * BSD 2-Clause License (https://www.opensource.org/licenses/bsd-license.php)
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ *      notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ *      copyright notice, this list of conditions and the following disclaimer
+ *      in the documentation and/or other materials provided with the
+ *      distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * You can contact the author at:
+ *   - xxHash homepage: https://www.xxhash.com
+ *   - xxHash source repository: https://github.com/Cyan4973/xxHash
+ */
+
+	.arch armv8-a+sve
+
+#include "asmdefs.h"
```

Comment:
This is an assembly file, not a C file with some assembly.
It means the compilation pipeline will be different for this file.

This make look like a minor detail, but I've experienced critical issues doing the same thing in other projects.

Probably the most critical one is the impact this can have on the executable stack flag, which then trigger security related issues.
It needs to be taken care of at compiler level.
This can be an annoyance to setup. You may refer to https://github.com/facebook/zstd/pull/2964/files#diff-a19342636d0d39fbbfc6ebf4e3839ed2ae4e043e7c9e37ca96226c0b7ad3a499R101 for an example.

Reply:
Does it mean that I could avoid executable stack flag if I don't include `asmdefs.h`?

Reply:
> Does it mean that I could avoid executable stack flag if I don't include `asmdefs.h`?

No, the issue is about `xxh_arm64sve.S` itself